### PR TITLE
fix(emqx_ee_connector): log reason for failure when starting influxdb connector

### PR DESCRIPTION
**Pull request by olcai**
The `influxdb-client-erl` lib doesn't return a reason for why it is not alive. This commit is built upon a PR, https://github.com/emqx/influxdb-client-erl/pull/44,  that allows the caller to request this extra information. We use it to log the error reason so that it is easier to debug connection issues to InfluxDB.

Fixes EMQX-8785.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/<version>/(feat|fix)-<PR-id>.en.md` and `.zh.md` files
- [X] For internal contributor: there is a jira ticket to track this change
- [ ] ~If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up~
- [ ] ~Schema changes are backward compatible~

### CODE REVIEW

File: lib-ee/emqx_ee_connector/rebar.config [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {erl_opts, [debug_info]}.
 {deps, [
   {hstreamdb_erl, {git, "https://github.com/hstreamdb/hstreamdb_erl.git", {tag, "0.2.5"}}},
-  {influxdb, {git, "https://github.com/emqx/influxdb-client-erl", {tag, "1.1.8"}}},
+  {influxdb, {git, "https://github.com/olcai/influxdb-client-erl", {branch, "add_reason_return_to_is_alive"}}},
```
Comment (olcai) [inline lib-ee/emqx_ee_connector/rebar.config:4]: This needs to point to the right repo and tag before merging.

File: changes/ee/fix-9881.zh.md [hunk 2]
```diff
@@ -0,0 +1 @@
+The InfluxDB connector will now log the reason for connection errors.
```
Comment (olcai) [inline changes/ee/fix-9881.zh.md:1]: Requires translation.

File: changes/ee/fix-9881.en.md [hunk 3]
```diff
@@ -0,0 +1 @@
+The InfluxDB connector will now log the reason for connection errors.
```
Comment (zmstone) [inline changes/ee/fix-9881.en.md:1]: In this pull request, we have enhanced the error logs related to InfluxDB connectivity health checks. 
Previously, if InfluxDB failed to pass the health checks using the specified parameters, the only message provided was "timed out waiting for it to become healthy".
With the updated implementation, the error message will be displayed in both the logs and the dashboard, enabling easier identification and resolution of the issue.

File: changes/ee/fix-9881.zh.md (hunk 2 above)
Comment (zmstone) [reply changes/ee/fix-9881.zh.md:1]: 增强了与 InfluxDB 连接健康检查相关的错误日志。
在此更改之前，如果使用配置的参数 InfluxDB 未能通过健康检查，用户仅能获得一个“超时”的信息。
现在，详细的错误消息将显示在日志和控制台，从而让用户更容易地识别和解决问题。

File: lib-ee/emqx_ee_connector/src/emqx_ee_connector_influxdb.erl [hunk 4]
```diff
@@ -249,13 +249,15 @@ do_start_client(
...
                     {ok, State};
-                false ->
-                    ?tp(influxdb_connector_start_failed, #{error => influxdb_client_not_alive}),
+                {false, Reason} ->
+                    ?tp(influxdb_connector_start_failed, #{
+                        error => influxdb_client_not_alive, reason => Reason
+                    }),
                     ?SLOG(warning, #{
-                        msg => "starting influxdb connector failed",
+                        msg => "starting influxdb connector failed - client is not alive",
```
Comment (zmstone) [inline lib-ee/emqx_ee_connector/src/emqx_ee_connector_influxdb.erl:257]: one of SLOG's goal is toe make `msg` more indexer friendly.
```suggestion
                        msg => "failed_to_start_influxdb_connector",
```

**Review by zmstone — APPROVED** [review]

**Pull request merged**